odoo/addons/base/models/ir_model.py

```diff
@@ -308,6 +337,15 @@ def _add_manual_models(self):
         # clean up registry first
         custom_models = [name for name, model_class in self.pool.items() if model_class._custom]
         for name in custom_models:
+            # remove the custom model from its parents' inheritance list
+            # otherwise registry might crash on next reload
+            parent_names = self.pool.models[name]._inherit
+            # custom models could inherit from mixins (`mail.thread`, `mail.activity.mixin`)
+            if isinstance(parent_names, list):
+                for parent_name in parent_names:
+                    self.pool.models[parent_name]._inherit_children.discard(name)
+            else:
+                self.pool.models[parent_names]._inherit_children.discard(name)
```

Comment:
Ouch!  I never thought about that one.
Maybe we should make a proper method for that on the registry (or implement `__delitem__`)

Reply:
yes, perhaps implementing `__delitem__` is better, but then we need to change the next statement to 
```py
del self.pool[name]`
```
since the models attribute on the registry is a simple dict

Reply:
i've done just that, could you check my changes? :smiley: